client/components/MenuBar.jsx

```diff
@@ -0,0 +1,196 @@
+    links: ["Bags & Backpacks","Hats ","Socks"]
+  },
+  {
+    category: "Shop Collection",
+    links: ["Rep Your City","Nike Adventure Club","Sandals & Slides","Surf & Swimwear","Jordan Kids","Fan Gear","Nike FlyEase"]
+  },
+]
+
+const menu5 = [
+  {
+    category: "New & Featured",
+    links: ["Customize with Nike By You","Nike By You New Releases","Basketball By You","Blazer By You","Air Max By You"]
+  },
+  {
+    category: "Mens",
+    links: ["Basketball","Lifestyle","Running","Football","Training & Gym","Soccer","Baseball","Skateboarding"]
+  },
+  {
+    category: "Womens",
+    links: ["Lifestyle","Running","Training & Gym","Basketball","Soccer","Softball","Skateboarding"]
+  },
+]
+
+const menu6 = [
+  {
+    category: "Sale - Up to 40% Off",
+    links: []
+  },
+  {
+    category: "Mens Sale",
+    links: ["Shoes","Clothing","Equipment"]
+  },
+  {
+    category: "Womens Sale",
+    links: ["Shoes","Clothing","Equipment"]
+  },
+  {
+    category: "Kids Sale",
+    links: ["Shoes","Clothing","Equipment"]
+  },
+]
+
+class MenuBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeMenu: null,
+    }
+  }
+
+  openMenu = (e) => {
+    const menuNumber = e.target.dataset.menu;
+    this.setState({
+      activeMenu: menuNumber
+    })
+  }
+
+  closeMenu = () => {
+    this.setState({
+      activeMenu: null
+    })
+  }
+
+  render() {
+    const { activeMenu } = this.state;
+    return (
+      <div className={styles.menuBar}>
+        <ul className={styles.navLinks}>
+          <li>
+            <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="1">New Releases</a>
+            <Menu items={menu1} menuNumber="1" activeMenu={activeMenu} />
+          </li>
+          <li>
+            <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="2">Men</a>
+            <Menu items={menu2} menuNumber="2" activeMenu={activeMenu} />
+          </li>
+          <li>
+            <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="3">Women</a>
+            <Menu items={menu3} menuNumber="3" activeMenu={activeMenu} />
+          </li>
+          <li>
+            <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="4">Kids</a>
+            <Menu items={menu4} menuNumber="4" activeMenu={activeMenu} />
+          </li>
+          <li>
+            <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="5">Customize</a>
+            <Menu items={menu5} menuNumber="5" activeMenu={activeMenu} />
+          </li>
+          <li>
+          <a className={styles.menuHeader} onMouseEnter={this.openMenu} onMouseLeave={this.closeMenu} data-menu="6">Sale</a>
+            <Menu items={menu6} menuNumber="6" activeMenu={activeMenu} />
```

Comment:
I see repeated codes and hard-coded values in here.  Will you be able to grow/shrink this list according to the list defined from above?  This is one area that changes a lot in the real world.  You have to make this easy to maintain as much as possible.

Hint - keep the data separate from the code.

Reply:
good idea. What do you suggest, keeping the data in its own file? For example, in JSON format.

Reply:
as for the hard-coded menus, I think that's a valid point, and I'll probably go back and try to refactor this later to be dynamic if I have time.